lib/teleterm/services/connectmycomputer/connectmycomputer.go

```diff
@@ -0,0 +1,255 @@
+	})
+	if err != nil {
+		return noCertsReloaded, trace.Wrap(err)
+	}
+
+	s.cfg.Log.Info("Reissuing certs.")
+	// ReissueUserCerts called with CertCacheDrop and a bogus access request ID in DropAccessRequests
+	// allows us to refresh the role list in the certs without forcing the user to relogin.
+	//
+	// Sending bogus request IDs is not documented but it is covered by tests. Refreshing roles based
+	// on the server state is necessary for tsh request drop to work.
+	//
+	// If passing bogus request IDs ever needs to be removed, then there are two options here:
+	// * Pass a wildcard instead. This will break setups where people use access requests to make
+	//   Connect My Computer work. Most users will probably not use access requests for that though.
+	// * Invalidate the stored certs somehow to force the user to relogin. If Connect makes a request
+	//   after role setup and [client.IsErrorResolvableWithRelogin] returns true for the error from
+	//   the response, Connect will ask the user to relogin.
+	//
+	// TODO(ravicious): Expand [auth.ServerWithRoles.GenerateUserCerts] to support refreshing role
+	// list without having to send a bogus request ID.
+	err = certManager.ReissueUserCerts(ctx, client.CertCacheDrop, client.ReissueParams{
+		RouteToCluster: cluster.Name,
+		// AccessRequests:     cluster.GetLoggedInUser().ActiveRequests,
+		DropAccessRequests: []string{fmt.Sprintf("bogus-request-id-%v", uuid.NewString())},
+	})
+	return RoleSetupResult{CertsReloaded: true}, trace.Wrap(err)
+}
+
+// syncResourceUpdate calls a function which updates the given resource and then waits until the
+// cache propagates the change.
+func (s *RoleSetup) syncResourceUpdate(ctx context.Context, accessAndIdentity AccessAndIdentity, resource types.Resource, updateFunc func(context.Context) error) error {
+	watcher, err := accessAndIdentity.NewWatcher(ctx, types.Watch{
+		Kinds: []types.WatchKind{
+			{Kind: resource.GetKind()},
+		},
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer watcher.Close()
+
+	// Wait for OpInit.
+	select {
+	case <-ctx.Done():
+		return trace.Wrap(ctx.Err(), "initializing watcher")
+	case <-watcher.Done():
+		return trace.Wrap(watcher.Error(), "initializing watcher")
+	case event := <-watcher.Events():
+		if event.Type != types.OpInit {
+			return trace.Errorf("unexpected event type %q received from resource watcher", event.Type)
+		}
+	}
+
+	err = updateFunc(ctx)
+	if err != nil {
+		return trace.Wrap(err, "calling update function")
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return trace.Wrap(ctx.Err(), "waiting for OpPut event for %v", resource.GetName())
+		case <-watcher.Done():
+			return trace.Wrap(watcher.Error(), "waiting for OpPut event for %v", resource.GetName())
+		case event := <-watcher.Events():
+			if event.Type != types.OpPut {
+				continue
+			}
+
+			if event.Resource.GetKind() == resource.GetKind() && event.Resource.GetMetadata().Name == resource.GetName() {
```

Comment:
I'm going to add a comment which says

> Kind + name combo is enough to uniquely identify a resource within a single cluster.

But let me know what concern we're trying to address because I don't fully get it – in this context isn't kind + name always enough? Or is it that you think it's not common knowledge and is worth leaving a comment?